client/src/app/pages/migration-waves/components/status-table.tsx

```diff
@@ -84,18 +84,17 @@ export const WaveStatusTable: React.FC<IWaveStatusTableProps> = ({
       getTdProps,
     },
   } = tableControls;
-  //
+
   const getTicketByApplication = (tickets: Ticket[], id: number = 0) =>
     tickets.find((ticket) => ticket.application?.id === id);
 
-  const getTicketIssue = (appId: number | undefined) => {
+  const getTicketIssue = (appId?: number) => {
     if (appId) {
       const ticket = getTicketByApplication(tickets, appId);
       if (ticket) {
-        const types = getTrackerTypesByProjectId(
-          trackers,
+        const types = useTrackerTypesByProjectId(
           ticket.tracker.name,
-          ticket?.parent
+          ticket.parent
         );
```

Comment:
This has the same issue we identified before with `useTrackerProjectsByTracker` where it is conditionally calling a hook. It should be renamed `useTicketIssue`, and the `useTrackerTypesByProjectId` should be called outside the `if (ticket)` block, which will require its params to be optional (internally passing `enabled: false` to useQuery if they are missing). Then make sure `useTicketIssue` is not being called in a conditional or callback (called at render time, every render).

Reply:
Thanks @mturley , I removed the hook from being under condition.